### PR TITLE
Allow custom partitioners to have their own configs

### PR DESCRIPTION
**Pull request by zzbennett**
Patch for issue #49

### CODE REVIEW

**Comment by zzbennett** [comment]
[clabot:check]

**Comment by zzbennett** [comment]
@ewencp sorry for the obnoxious ping. Our use case requires us to be able to pass configs down to our custom partitioner so we'd really like to get this patch through. I'd be happy to modify or rewrite it in whatever way makes the most sense, so long as we have some way of getting the configs to our custom code.

File: kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java [hunk 1]
```diff
@@ -143,7 +143,17 @@ public void open(Collection<TopicPartition> partitions) {
             config.getClass(PartitionerConfig.PARTITIONER_CLASS_CONFIG);
 
     Partitioner<FieldSchema> partitioner = partitionerClass.newInstance();
-    partitioner.configure(new HashMap<>(config.plainValues()));
+
+    HashMap<String, Object> plainValues = new HashMap<>(config.plainValues());
+    HashMap<String, ?> originals = new HashMap<>(config.originals());
```
Comment (zzbennett) [inline kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java]: oh whoops, didn't mean to do that. will update

Comment (zzbennett) [inline kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java]: Okay no I remember why I did this. I was running into some weirdness with the Capture<?> values in the Hashmap. I'll see if I can find a more sensible workaround

**Comment by zzbennett** [comment]
Committed too soon, still working on it, will update with another tweak

**Comment by zzbennett** [comment]
@kkonstantine updated with feedback. Thanks for taking a look!

**Comment by kkonstantine** [comment]
retest this please

**Comment by kkonstantine** [comment]
ok to test

**Comment by kkonstantine** [comment]
Great thanks! (I'm a bit surprised that checkstyle didn't catch missing whitespace after `if` and `for`. No big deal, I'll fix before merging.)

**Comment by kkonstantine** [comment]
Merged manually. Closing here.

**Pull request closed without merging**